acstools/utils_findsat_mrt.py

```diff
@@ -1199,10 +1200,75 @@ def update_dq(filename, ext, mask, dqval=16384, verbose=True):
         tailored for ACS/WFC.
     verbose : bool, optional
         Print extra information to the terminal.
+    expand_mask: bool, optional
+        Allows the mask to be expanded to match the size of the original mask.
+        This is relevant for masks that were generated used binned versions of
+        the original image.
 
     """
     with fits.open(filename, mode='update') as pf:
         dqarr = pf[ext].data
+
+        # check that the old mask and new mask are the same size
+        same_size = mask.shape == dqarr.shape
+        mask_larger = np.any([mask.shape[i] > dqarr.shape[i]
+                              for i in range(mask.ndim)])
+
+        # provide informative error messages depending on mask/dqarr size
+        if not same_size:
+            LOG.info('Inconsistent mask sizes: \n'
+                     'Input mask: {} \n'
+                     'DQ array: {}'.format(mask.shape, dqarr.shape))
+
+            if mask_larger:
+                LOG.error('Mask is larger than the DQ array in one or more '
+                          'dimensions.\nThis routine cannot currently rescale'
+                          'masks that are larger than the origin input image')
+                return
+
+            elif not mask_larger and not expand_mask:
+                LOG.error('Cannot proceed due to size mismatch.\n'
+                          'Set expand_mask=True if the mask was generated '
+                          'with binned data and needs to be enlarged to the '
+                          'original size.')
+                return
+            else:
+                LOG.info('Enlarging mask to original size.')
+
+                # Determine factor by which to scale-up image. Scales should
+                # be the same for the x and y dimensions, after we round down
+                # to the nearest integer in case there are extra rows/columns
+                # that do not divide evenly. These extra rows/columns will be
+                # added back at the end. This approach is the reverse of
+                # astropy.nddata.block_reduce, which trims extra rows/cols
+                # first.
+
+                factor_x = np.floor(dqarr.shape[1]/mask.shape[1]).astype(int)
+                factor_y = np.floor(dqarr.shape[0]/mask.shape[0]).astype(int)
+                if factor_x != factor_y:
+                    raise ValueError('Mask dimensions require scaling by'
+                                     'different amounts in order to match the'
+                                     'DQ array but this is not allowed.')
```

Comment:
need spaces between 'by' and 'different, 'the' and 'DQ'

Reply:
I tried a square mask (1000x1000), and got this error. I think the error could be more helpful, but I'm not sure how. Perhaps something to do with the aspect ratio of the mask must be close to that of the DQ array?

Reply:
thanks Jenna. Do you meant the error message itself could be more helpful? It could say something like "Mask aspect ratio too different from DQ array" for instance.

Reply:
Yes, that's what I mean. Essentially, the message should help the user figure out how to fix the problem. Could think about printing out the aspect ratio of the mask and the dq array, and tell the user to make them more similar?

Reply:
Ok, I modified that error message. Let me know what you think

Reply:
Looks good to me!